`loader` flag for esbuild plugin

nicholaides opened the pull request:
## Adds a `loader` flag to the esbuild plugin.
This flag can be used to tell esbuild that once a .hera file is compiled it should be treated as a .ts file (or whatever you want) and continue to be processed by loaders.

e.g.
```ts
esbuild.build({
  ...
  heraPlugin({ loader: "ts" })
})
```

## ~~Stops using default export (❗️API change)~~
~~I couldn't figure out how to get the default export working in both CJS and ESM (e.g. the mocha test suite)~~

## Also
- add mocha tests for the esbuild plugin
- add smoke test (`yarn test:esbuild-plugin`) that ensure the esbuild plugin is built and exported correctly
- rewrite .js files' imports from .civet to .js. Previously this was only applied to .civet.d.ts iles and ./dist/hera

STRd6 commented on source/esbuild-plugin.civet:
```diff
@@ -1,19 +1,21 @@
+export interface HeraPluginOptions < CompilerOptions
+  loader?: Loader
+
+export function heraPlugin({ loader, ...heraOptions }: HeraPluginOptions = {}): Plugin
```
We could also add `export default heraPlugin` below this to allow for both styles of import in ESM. CJS may still need to do something like `require('hera/esbuild-plugin').default` when requiring but it's not a huge deal.

edemaine replied:
https://github.com/DanielXMoore/Civet/blob/0d97621082c7c0d2201d57498816b67ad95e8838/build/esbuild.civet#L94 should probably fix the latter issue.

nicholaides replied:
@STRd6 It seems like @edemaine's suggestion is the best way to not change the API. But if you are ok with changing the API should we switch to using a named export instead of default exports?

STRd6 replied:
I generally prefer default exports, that way consumers of the package don't have to keep track of an arbitrary intermediary name. If we also want to provide a named export in addition to the default that's fine too.

nicholaides commented:
I implemented @edemaine's suggestion for fixing CJS default export for the esbuild-plugin. It works great. 

I also added test scripts that verify that the default exports work in both import and require contexts. I wanted a test in there because it seems like an easy thing to break without realizing it.
